test_run.py

```diff
@@ -338,13 +339,17 @@ def release_urls(self, name, version):
     assert basename == "whatever.tar.gz"
 
 
-def test_download_package_whl(monkeypatch):
+def test_download_package_whl(monkeypatch, mocker):
     def fake_urlretrieve(url, basename):
         assert url == "/path/to/myplugin-1.0.0-py2.py3-none-any.whl"
         assert basename == "myplugin-1.0.0-py2.py3-none-any.whl"
 
     monkeypatch.setattr("run.urlretrieve", fake_urlretrieve)
 
+    import wheel.wheelfile
+    m = mocker.patch.object(wheel.wheelfile, 'WheelFile', autospec=True)
+    m.return_value.compatible = True
+
```

Comment:
@nicoddemus  Why the inline imports, instead of just patching in the namespace `run.WheelFile`?

Reply:
Personal preference only, really. You see a problem with it?

Reply:
Yeah. From the [style guide](https://www.python.org/dev/peps/pep-0008/?#imports):

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

A module should be explicit about the dependencies, I guess?  You don't want any import error triggered later in runtime when `download_package` is called, you want it upfront as soon as `run.py` is loaded.

Reply:
No worries, done!